go/vt/vttablet/tabletserver/messager/cache.go

```diff
@@ -99,13 +110,14 @@ func (mc *cache) Add(mr *MessageRow) bool {
 		return false
 	}
 	id := mr.Row[0].ToString()
-	// Don't check for nil. Messages that are popped for
-	// send are nilled out.
-	if _, ok := mc.messages[id]; ok {
+	if _, ok := mc.inFlight[id]; ok {
```

Comment:
Can you please use `bool` as value type for this map? Then you have a set instead of a map and you can simplify this code for example:

```go
if mc.inFlight[id] {
```